### PR TITLE
added JobCard and JobCardList

### PR DESCRIPTION
**Pull request by ryan280**

### CODE REVIEW

File: client/src/components/JobCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+export const JobCard = (props: any) => {
+    // Constants
+    const BASE_CLEARBIT_URL = 'https://logo.clearbit.com/';
+
+    // State variables
+    const [url, setUrl] = useState(props.url);
+    const [company, setCompany] = useState(props.company);
+    const [role, setRole] = useState(props.role);
+    const [color, setColor] = useState(props.color);
```
Comment (bwormser) [inline client/src/components/JobCard/index.tsx]: Are these redundant if the useEffect also sets them to the same value?

Comment (ryan280) [reply client/src/components/JobCard/index.tsx]: The use effect is to update them if the value changes

**Comment by mpolimen** [comment]
Too chubby

**Comment by mpolimen** [comment]
Add accordion

File: client/src/components/JobCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        setColor(props.color);
+    }, [props.color]);
+
+     // NOTE: I dont know a way to use one Link tag and keep the styling the same
+     return (
+        <Grid item xs={12}>
+            <Link to={'/'} style={{textDecoration: 'none'}}>
+                <Card className='card' style={{ backgroundColor: color }}>
+                    <Box sx={{ display: 'flex', flexDirection: 'column', margin: '10px'}}>
+                        <CardMedia
```
Comment (GraceMattern) [inline client/src/components/JobCard/index.tsx:42]: Logo is not responsive

File: client/src/components/JobCardList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    //     setUrl('google.com');
+    //     setCompany('Google');
+    //     setRole('Software Engineer');
+    //     setColor('#F6C92E');
+
+    // }, []);
+
+    return(
+        <Grid container spacing={2} direction='column' justifyContent='center' alignItems='stretch' style={{ margin: '20px', width: '20%', border: '2px solid black'}}> Applied
```
Comment (GraceMattern) [inline client/src/components/JobCardList/index.tsx]: This is a grid but not a dropdown

**Review by GraceMattern — COMMENTED** [review]
Looks mostly good but I would recommend you remove the blocks of comments unless we will be using it for later development

File: client/src/components/JobCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import './style.css'
+import { useEffect, useState } from "react";
+import { Link } from 'react-router-dom';
+import { Grid, Box, Card, CardMedia, CardContent, Typography } from '@material-ui/core';
+import { Margin } from '@mui/icons-material';
+
+export const JobCard = (props: any) => {
```
Comment (roccopolimen) [inline client/src/components/JobCard/index.tsx]: do not export const, you need to export default like how we do it elsewhere so the importing can be done the same way.

Comment (roccopolimen) [inline client/src/components/JobCard/index.tsx]: :any is useless, if you are using types, declare a PropType at the top of this file to be used by it.

File: client/src/components/JobCard/index.tsx (hunk 1 above)
Comment (roccopolimen) [reply client/src/components/JobCard/index.tsx]: I also believe this is redundant

File: client/src/components/JobCard/index.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import './style.css'
+import { useEffect, useState } from "react";
+import { Link } from 'react-router-dom';
+import { Grid, Box, Card, CardMedia, CardContent, Typography } from '@material-ui/core';
```
Comment (roccopolimen) [inline client/src/components/JobCard/index.tsx]: MUI component imports should be `from '@mui/material`
check slack for my message on this

File: client/src/components/JobCard/index.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                        <Typography className='role' variant="subtitle1" component="div">
+                            {role}
+                        </Typography>
+                    </CardContent>
+                </Card>
+            </Link>
+        </Grid>
+      );
+
+    // return (
```
Comment (roccopolimen) [inline client/src/components/JobCard/index.tsx]: remove dead code?

File: client/src/components/JobCardList/index.tsx (hunk 3 above)
Comment (roccopolimen) [reply client/src/components/JobCardList/index.tsx]: +1

**Review by roccopolimen — COMMENTED** [review]
Some changes needed before approval

File: client/src/components/JobCard/index.tsx (hunk 4 above)
Comment (ryan280) [reply client/src/components/JobCard/index.tsx]: Please elaborate if you want me to create a new PropType in the typings file or if there is a declared one already. I do not see one currently.

**Comment by ryan280** [comment]
Fixed many errors and chubbiness. Be aware it is meant to fit to the size of its container, so if it place it on the landing page it will fit edge to edge.  On the upcoming page it will be placed in a grid container presumably that will contain it to a desirable size.

**Comment by GraceMattern** [comment]
Looks better. I think you need to pull from main though before merging

**Comment by mpolimen** [comment]
Figma: image
Your card: 
image

It doesn't have to be an exact, like take some creative liberty, but this looks like dog water rn. Put more effort into this

**Comment by mpolimen** [comment]
This entire page has like no margins anywhere, use some: 
image

**Comment by mpolimen** [comment]
Cards look good, but are huge in comparison to other things (scale down): 
image

File: client/src/components/JobCard/index.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    // Constants
+    const BASE_CLEARBIT_URL = 'https://logo.clearbit.com/';
+
+    // State variables
+    const [url, setUrl] = useState('');
+    const [company, setCompany] = useState('');
+    const [role, setRole] = useState('');
+    const [color, setColor] = useState('');
+
+    useEffect(() => {
```
Comment (mpolimen) [inline client/src/components/JobCard/index.tsx:16]: Using a useEffect here, for these, does not make sense.

File: client/src/components/JobCard/index.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        setRole(props.role);
+    }, [props.role]);
+
+    useEffect(() => {
+        setColor(props.color);
+    }, [props.color]);
+
+     // NOTE: I dont know a way to use one Link tag and keep the styling the same
```
Comment (mpolimen) [inline client/src/components/JobCard/index.tsx:36]: Look into useNavigate, it's been used in our code base a couple times now 👍

File: client/src/components/JobCardList/index.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
+// import './style.css'
+import { useEffect, useState } from "react";
+// import { Link } from 'react-router-dom';
+import { Accordion, AccordionDetails, AccordionSummary, Grid, Typography } from '@mui/material';
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import JobCard from 'components/JobCard';
+
+const JobCardList = (props: any) => {
```
Comment (mpolimen) [inline client/src/components/JobCardList/index.tsx]: props cannot be of "any" type

File: client/src/components/JobCardList/index.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+                id="panel1a-header">
+                <Typography>{title}</Typography> 
+            </AccordionSummary>
+            <AccordionDetails>
+                <JobCard url={'google.com'} company={'Google'} role={'Software Engineer'} color={'#F6C92E'}/>
+                <JobCard url={'spotify.com'} company={'Spotify'} role={'Data Scientist'} color={'#76D27C'}/> 
+                <JobCard url={'netflix.com'} company={'Netflix'} role={'Software Engineer'} color={'#ED717D'}/> 
+            </AccordionDetails>
+        </Accordion>
+        // <Grid container spacing={2} direction='column' justifyContent='center' alignItems='stretch' style={{ margin: '20px', width: '20%', border: '2px solid black'}}> Applied
```
Comment (mpolimen) [inline client/src/components/JobCardList/index.tsx]: Should this commented out section be deleted

File: client/src/components/UpcomingBox/index.tsx [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+    // State variables
+    const [url, setUrl] = useState('');
+    const [title, setTitle] = useState('');
+    const [company, setCompany] = useState('');
+    const [role, setRole] = useState('');
+    const [date, setDate] = useState('');
+    
+
+    useEffect(() => {
```
Comment (mpolimen) [inline client/src/components/UpcomingBox/index.tsx:20]: Same issue as above

File: client/src/pages/Upcoming/index.tsx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+import JobCardList from "components/JobCardList";
+import './style.css';
+
+const Upcoming = () => {
+    return (
+        <>
+            <Slide direction='down' in={true} timeout={800}>
+                <AppBar position='sticky' className='AppBar' color='inherit'>
+                    <AppToolbar>
+                    <div className='HeaderContainer'>
```
Comment (mpolimen) [inline client/src/pages/Upcoming/index.tsx]: The header does not belong on this page...
The drawer will eventually be here, but we're distinctly not using a nav bar for our site, like we've spoke about

**Comment by mpolimen** [comment]
Lots to improve on... Moving the trello card back to in progress

File: client/src/pages/Upcoming/index.tsx (hunk 12 above)
Comment (GraceMattern) [reply client/src/pages/Upcoming/index.tsx]: +1

**Comment by GraceMattern** [comment]
I agree with many of the comments Marco made. Also there is some element on the upcoming page that is making the window longer than the typical 9:16 dimension. I think it is regarding your Grid for the JobCardList (line 47 in pages/Upcoming)

File: client/src/components/JobCard/index.tsx (hunk 7 above)
Comment (ryan280) [reply client/src/components/JobCard/index.tsx:16]: So what would make sense?

**Comment by ryan280** [comment]
Would like a second opinion on the cards being "Huge" I really like the way they look

**Comment by roccopolimen** [comment]
I think cards look kinda thick but idk how much I care about it as long as:
* it looks fine responsively, and in unison with the rest of the page
if that is guaranteed then good, but idk if it is I havent seen the full page all together to approve it

File: client/src/components/JobCard/index.tsx (hunk 7 above)
Comment (mpolimen) [reply client/src/components/JobCard/index.tsx:16]: Once the data comes in from the backend fetch call, use one useEffect to update all variables (if you even need to have these variables separated, I'm not sure)

**Review by mpolimen — COMMENTED** [review]
Let's put it all together with the api hooks and link it to the backend. That was the expectation for this week, and it's not done yet.

**Comment by ryan280** [comment]
Navigate to /dashboard from login/signup page. If there isn't a current cycle useNavigate to Brian's first cycle page

**Comment by roccopolimen** [comment]
assuming it works as intended since I saw some of it in action. 

The Job Position in the right side collapsible is still a gray color that will not pass tota11y

mobile does not look great, but it is servicable
same with the modal for adding a job

**Comment by mpolimen** [comment]
WARNING in src/components/SideDrawer/index.tsx
  Line 1:17:  'useState' is defined but never used                @typescript-eslint/no-unused-vars
  Line 1:27:  'useEffect' is defined but never used               @typescript-eslint/no-unused-vars
  Line 60:9:  'handleSignOut' is assigned a value but never used  @typescript-eslint/no-unused-vars

src/components/UpcomingBox/index.tsx
  Line 4:21:  'Card' is defined but never used  @typescript-eslint/no-unused-vars

src/pages/OfferDash/index.tsx
  Line 13:5:  'CircularProgress' is defined but never used  @typescript-eslint/no-unused-vars
  Line 14:5:  'Box' is defined but never used               @typescript-eslint/no-unused-vars

src/pages/Upcoming/index.tsx
  Line 6:10:    'AppBar' is defined but never used
                   @typescript-eslint/no-unused-vars
  Line 6:31:    'Card' is defined but never used
                   @typescript-eslint/no-unused-vars
  Line 6:37:    'CardContent' is defined but never used
                   @typescript-eslint/no-unused-vars
  Line 6:50:    'CardMedia' is defined but never used
                   @typescript-eslint/no-unused-vars
  Line 9:10:    'Save' is defined but never used
                   @typescript-eslint/no-unused-vars
  Line 9:21:    'Delete' is defined but never used
                   @typescript-eslint/no-unused-vars
  Line 17:12:   'changed' is assigned a value but never used
                   @typescript-eslint/no-unused-vars
  Line 32:18:   'applicationData' is assigned a value but never used
                   @typescript-eslint/no-unused-vars
  Line 32:46:   'applicationIsLoading' is assigned a value but never used
                   @typescript-eslint/no-unused-vars
  Line 32:77:   'applicationIsError' is assigned a value but never used
                   @typescript-eslint/no-unused-vars
  Line 114:62:  Array.prototype.map() expects a value to be returned at the end of arrow function  array-callback-return
  Line 127:62:  Array.prototype.map() expects a value to be returned at the end of arrow function  array-callback-return

**Comment by mpolimen** [comment]
Events that are checked as complete should not show up in the upcoming page (or should have some indicator that it is complete)

**Comment by mpolimen** [comment]
Put some space between the logo and the event title

**Comment by mpolimen** [comment]
Made changes that I'm happy with; as far as I can tell, it works now 👍 so consider this a LGTM

**Comment by ryan280** [comment]
Good changes LGTM thank you

**Pull request merged**